### PR TITLE
Update ExecutionResult reporting.

### PR DESCRIPTION
**Pull request by zmrocze**
- pretty print execution result on 'shouldSucced' etc. fails
 - turn on colored test output

### CODE REVIEW

File: cabal.project [hunk 1]
```diff
@@ -6,3 +6,5 @@ tests: true
 benchmarks: true
 
 allow-newer: size-based:template-haskell
+
+test-show-details: direct
```
Comment (mikekeke) [inline cabal.project:10]: Newline missing

**Comment by zmrocze** [comment]
Before the `shouldSucced` predicate used `Show` on `ExecutionResult`. This also outputed contract logs, but in an accidental way, using derived `Show`.

Now `shouldSucced` (/`shouldFail`) prints only final contract state and execution outcome on failure (smaller output). Printing logs and budget stats can be made using `assertExecutionWith`. Before user defaulted to seeing these from `Show` output, which looks worse than `Pretty`. Now he might not know that there's this feature so an additional line of log is added. Namely every failing test of type `TestContract` prints "Use assertExecutionWith to show contract logs or budgets." 

Coloring is turned on with cabal flag: green checkmarks on success and red everything on fail.
Before 
```
     FAIL
        But it didn't.
        Result: ExecutionResult {outcome = Left (CaughtException This Exception was thrown intentionally in Contract.
        
        CallStack (from HasCallStack):
          error, called at test/Spec/TestContract/SimpleContracts.hs:48:20 in main:Spec.TestContract.SimpleContracts), txStats = ContractStats {estimatedBudgets = fromList []}, contractState = (), contractLogs = LogsList {getLogsList = [LogLine {logLineContext = ContractLog, logLineLevel = Info {ltLogTypes = [AnyLog]}, logLineMsg = Some contract log with Info level.},LogLine {logLineContext = ContractLog, logLineLevel = Debug {ltLogTypes = [AnyLog]}, logLineMsg = Another contract log with debug level.}]}}
        Use -p '/Contract 3.Contract should succeed/' to rerun this test only.

```
After
```
      Contract should succeed:                                                                           FAIL
        But it didn't.
        Result: Execution result {
            outcome: Left (CaughtException This Exception was thrown intentionally in Contract.
        
            CallStack (from HasCallStack):
              error, called at test/Spec/TestContract/SimpleContracts.hs:48:20 in main:Spec.TestContract.SimpleContracts)
            final state:  ()
        }
        
        Use assertExecutionWith to show contract logs or budgets.
        Use -p '/Contract 3.Contract should succeed/' to rerun this test only.

```
and it's red

**Comment by mikekeke** [comment]
>  Now he might not know that there's this feature so an additional line of log is added. Namely every failing test of type TestContract prints "Use assertExecutionWith to show contract logs or budgets."

👍

**Review by mikekeke — APPROVED** [review]

**Comment by mikekeke** [comment]
LGTM!

**Review by IAmPara0x — APPROVED** [review]

**Comment by zmrocze** [comment]
ill just add that newline in cabal and - if we already have it - will add aboves to the commit description.

**Pull request merged**